src/client/dfs/dfs.c

```diff
@@ -325,16 +328,17 @@ fetch_entry(daos_handle_t oh, daos_handle_t th, const char *name,
 	rc = daos_obj_fetch(oh, th, &dkey, akeys_nr, iods, sgls, NULL, NULL);
 	if (rc) {
 		D_ERROR("Failed to fetch entry %s (%d)\n", name, rc);
-		return rc;
+		D_GOTO(out, rc);
```

Comment:
I thought the consensus was to get rid of D_GOTO?

Reply:
there was? it's still used all over the place, at least in most of the client side..

Reply:
It's a mess.   I never really understood the reasoning to remove it but I thought there was a decision.   We should probably revisit it and pick one side or the other and just do a blanket removal (or insertion) everywhere.

One disadvantage is tools like coverity can't seem to handle it so they report unused labels.

Reply:
But I actually like D_GOTO because it's a clean way to set the rc and jump in one line